lib/src/table/macos_table_datasource.dart

```diff
@@ -0,0 +1,146 @@
+import 'dart:async';
+
+import 'package:macos_ui/src/library.dart';
+import 'package:macos_ui/src/table/macos_table_columns.dart';
+import 'package:macos_ui/src/table/macos_table_order.dart';
+import 'package:macos_ui/src/table/macos_table_selection.dart';
+
+/// A combination of a row [value] and a persistent identifier [key] for the row.
+@immutable
+class MacosTableValue<T> {
+  const MacosTableValue({
+    required this.key,
+    required this.value,
+  });
+
+  /// A persistent identifier for this row.
+  ///
+  /// If you select a row and change the data or order,
+  /// the key is how the table knows which row was selected.
```

Comment:
If you could rewrite this without the use of "you", that would be great - something along the lines of "The key allows the table to know which row has been selected". It's descriptive without addressing the reader directly.